exceptions/pulp/exceptions/base.py

```diff
@@ -1,93 +1,44 @@
-            'description': str(self),
-            'data': self.error_data,
-            'sub_errors': []
-        }
-        for error in self.child_exceptions:
-            if isinstance(error, PulpException):
-                result['sub_errors'].append(error.to_dict())
-            else:
-                result['sub_errors'].append({'code': 'PLP0000',
-                                             'description': str(error),
-                                             'data': {},
-                                             'sub_errors': []})
-        return result
-
-    def __str__(self):
-        class_name = self.__class__.__name__
-        msg = _('Pulp exception occurred: %(c)s') % {'c': class_name}
-        if self.args and isinstance(self.args[0], basestring):
-            msg = self.args[0]
-        return msg.encode('utf-8')
-
-    def data_dict(self):
-        return {'args': self.args}
-
-
-class PulpCodedException(PulpException):
-    """
-    Base class for exceptions that put the error_code and data as init arguments
-    """
-    def __init__(self, error_code=error_codes.PLP0001, **kwargs):
-        super(PulpCodedException, self).__init__()
+        if not isinstance(error_code, basestring):
+            raise RuntimeError("Error code must be an instance of basestring.")
         self.error_code = error_code
-        if kwargs:
-            self.error_data = kwargs
-        # Validate that the coded exception was raised with all the error_data fields that
-        # are required
-        for key in self.error_code.required_fields:
-            if key not in self.error_data:
-                raise PulpCodedException(error_codes.PLP0008, code=self.error_code.code,
-                                         field=key)
 
     def __str__(self):
-        msg = self.error_code.message % self.error_data
-        return msg.encode('utf-8')
+        """
+        Returns the string representation of the exception.
+
+        Each concrete class that inherits from :class: `pulp.server.exception.PulpException` is
+        expected to implement it's own __str__() method.
+        """
+        pass
 
 
-class PulpExecutionException(PulpException):
+def exception_to_dict(exc, traceback):
     """
-    Base class of exceptions raised during the execution of Pulp.
+    Return a dictionary representation of an Exception.
 
-    This class should be used as a graceful server-side error while running
-    an operation. It is acceptable to instantiate and use this class directly.
+    :param exc: Exception that is being serialized
+    :type exc: Exception
+    :param traceback:
+    :type traceback:
 
-    Subclasses to this exception can be used to further describe any problems
-    encountered by the server.
+    :return: dictionary representing the Exception
+    :rtype: dict
     """
-    # NOTE intermediate exception class, no overrides will be provided
-    pass
+    result = {'code': None, 'description': str(exc), 'traceback': traceback}
```

Comment:
The traceback is already a string at this point. We get the value of it from this[0] attribute.
[0] http://docs.celeryproject.org/en/latest/internals/reference/celery.datastructures.html#celery.datastructures.ExceptionInfo.traceback